test/e2e/atlas/kubernetes_config_generate_test.go

```diff
@@ -63,44 +188,89 @@ func getK8SEntities(data []byte) ([]runtime.Object, error) {
 }
 
 func TestKubernetesConfigGenerate(t *testing.T) {
-	g := newAtlasE2ETestGenerator(t)
-	g.generateProjectAndCluster("importer-test")
-
 	cliPath, err := e2e.AtlasCLIBin()
 	require.NoError(t, err)
 
 	n, err := e2e.RandInt(255)
 	require.NoError(t, err)
 
-	targetNamespace := fmt.Sprintf("importer-namespace-%d", n)
+	targetNamespace := "importer-namespace"
+
+	projectName := fmt.Sprintf("test-project-%s", n)
+	clusterReplicaSetName := fmt.Sprintf("test-cluster-replicaset-%s", n)
+	clusterServerlessName := fmt.Sprintf("test-cluster-serverless-%s", n)
+
+	helper := NewK8sHelper(t)
+	helper.NewProject(&atlas.Project{Name: projectName, OrgID: helper.orgID})
+	helper.NewCluster(&atlas.Cluster{
+		BiConnector: &atlas.BiConnector{
+			Enabled: pointers.MakePtr(true),
+		},
+		ClusterType:           "REPLICASET",
+		DiskSizeGB:            pointers.MakePtr[float64](10),
+		GroupID:               helper.project.ID,
+		Name:                  clusterReplicaSetName,
+		ProviderBackupEnabled: pointers.MakePtr(false),
+		ProviderSettings: &atlas.ProviderSettings{
+			InstanceSizeName: "M10",
+			ProviderName:     "AWS",
+			RegionName:       "US_EAST_1",
+		},
+		ReplicationFactor: pointers.MakePtr[int64](3),
+		ReplicationSpecs: []atlas.ReplicationSpec{
+			{
+				NumShards: pointers.MakePtr[int64](1),
+				ZoneName:  "Zone 1",
+				RegionsConfig: map[string]atlas.RegionsConfig{
+					"US_EAST_1": {
+						Priority:       pointers.MakePtr[int64](7),
+						AnalyticsNodes: pointers.MakePtr[int64](0),
+						ElectableNodes: pointers.MakePtr[int64](3),
+						ReadOnlyNodes:  pointers.MakePtr[int64](0),
+					},
+				},
+			},
+		},
+	})
+	helper.NewServerlessInstance(&atlas.ServerlessCreateRequestParams{
+		Name: clusterServerlessName,
+		ProviderSettings: &atlas.ServerlessProviderSettings{
+			BackingProviderName: "AWS",
+			ProviderName:        "SERVERLESS",
+			RegionName:          "US_EAST_1",
+		},
+	})
+
+	time.Sleep(30 * time.Second)
```

Comment:
we have a built in command `atlas cluster watch` that waits for the cluster to be ready. The logic is already in `deployClusterForProject`